helm/loki/templates/_helpers.tpl

```diff
@@ -286,3 +287,153 @@ http {
   }
 }
 {{- end }}
+
+{{/*
+/!\ Giantswarm override to work around a bug in azure where `use_federated_token` fails.
+/!\ Added on chart 4.10.0 / Loki 2.7.2
+Generated storage config for loki common config
+*/}}
+{{- define "loki.commonStorageConfig" -}}
+{{- if .Values.minio.enabled -}}
+s3:
+  endpoint: {{ include "loki.minio" $ }}
+  bucketnames: {{ $.Values.loki.storage.bucketNames.chunks }}
+  secret_access_key: {{ $.Values.minio.rootPassword }}
+  access_key_id: {{ $.Values.minio.rootUser }}
+  s3forcepathstyle: true
+  insecure: true
+{{- else if eq .Values.loki.storage.type "s3" -}}
+{{- with .Values.loki.storage.s3 }}
+s3:
+  {{- with .s3 }}
+  s3: {{ . }}
+  {{- end }}
```

Comment:
I am not a big fan of this kind of syntax, I would prefer to use something more readable if possible

```suggestion
  s3: {{ .s3 }}
```

If accepted then please change all other section using `with`

Reply:
It's part of the original `loki.commonStorageConfig`, I prefer to not diverge too much.
If we're not happy with this syntax we'd rather propose a fix upstream.

Reply:
Ok, fine with me.

Reply:
Also, I think that if you do s3: in the config loki will fail. This could be changed with an if then but that's it